internal/broker/broker.go

```diff
@@ -398,86 +398,85 @@ func (b *Broker) IsAuthenticated(sessionID, authenticationData string) (string,
 	if err = b.updateSession(sessionID, session); err != nil {
 		return AuthDenied, "", err
 	}
-	return access, data, nil
+
+	// The data type is unknown, so we need to directly call its marshal function rather than relying
+	// on json.Marshal, which would reflect the struct fields and use what is implemented for the basic types.
+	encoded, err := data.MarshalJSON()
+	if err != nil {
+		return AuthDenied, "", fmt.Errorf("could not marshal data: %v", err)
```

Comment:
marshal is a little bit generic, I would that could not convert data to json or something like this?